test/index.js

```diff
@@ -0,0 +1,296 @@
+        nock('https://test.host.com')
+            .post('/api/auth')
+            .reply(200)
+
+            .get(`/api/job/${testJobId}`)
+            .reply(200, {
+                id: testJobId
+            });
+
+        const ST = Servicetrade(testOptions);
+        await ST.login();
+        const jobResponse = await ST.get(`job/${testJobId}`);
+        assert.deepEqual(jobResponse, null);
+    });
+});
+
+describe('Put tests', function() {
+
+    it('put job item success', async function() {
+        nock('https://test.host.com')
+            .post('/api/auth')
+            .reply(200)
+
+            .put(`/api/jobitem/${jobItemId}`, { libitemId: 9876 })
+            .reply(200, { data: { id: 1234 } });
+
+        const ST = Servicetrade(testOptions);
+        const jobItemResponse = await ST.put(`/jobitem/${jobItemId}`, { libitemId: 9876 });
+        assert.deepEqual(jobItemResponse.id, 1234);
+    });
+
+    it('return empty response cause not data property', async function() {
+        nock('https://test.host.com')
+            .post('/api/auth')
+            .reply(200)
+
+            .put(`/api/jobitem/${jobItemId}`, { libitemId: 9876 })
+            .reply(200, { id: 1234 });
+
+        const ST = Servicetrade(testOptions);
+        const jobItemResponse = await ST.put(`/jobitem/${jobItemId}`, { libitemId: 9876 });
+        assert.deepEqual(jobItemResponse, null);
+    });
+});
+
+describe('Post tests', function() {
+    const postData = {
+        quantity: 5,
+        cost: 6,
+        serviceLineId: 33,
+        name: 'fancy foo',
+        jobId: 33445566,
+        source: { type: 'refnumber', value: 'PO CC-654' }
+    };
+
+    it('post job item success', async function() {
+        nock('https://test.host.com')
+            .post('/api/auth')
+            .reply(200)
+
+            .post('/api/jobitem', postData)
+            .reply(200, { data: { id: 444 } });
+
+        const ST = Servicetrade(testOptions);
+        const jobItemResponse = await ST.post(`/jobitem`, postData);
+        assert.deepEqual(jobItemResponse.id, 444);
+    });
+
+    it('return empty response cause not data property', async function() {
+        nock('https://test.host.com')
+            .post('/api/auth')
+            .reply(200)
+
+            .put(`/api/jobitem/${jobItemId}`, { libitemId: 9876 })
+            .reply(200, { id: 1234 });
+
+        const ST = Servicetrade(testOptions);
+        const jobItemResponse = await ST.put(`/jobitem/${jobItemId}`, { libitemId: 9876 });
+        assert.deepEqual(jobItemResponse, null);
+    });
+});
+
+describe('Delete tests', function() {
+    const testJobId = 100;
+    it('delete job success', async function() {
+        nock('https://test.host.com')
+            .delete(`/api/job/${testJobId}`)
+            .reply(200, {});
+
+        const ST = Servicetrade(testOptions);
+        const response = await ST.delete(`/job/${testJobId}`);
```

Comment:
shouldn't there be a `login` call here (and a corresponding post to `/api/auth`)?